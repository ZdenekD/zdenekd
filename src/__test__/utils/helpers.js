import React from 'react';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';

/**
 * Factory function to create a shallow wrapper for App component
 * @function setup
 * @param {ReactComponent} Component - React component to render with shallow
 * @param {Object} props - Component props specific to this setup
 * @param {Object} state - Initial state for setup
 * @returns {ShallowWrapper}
 */
export const factory = (Component, props = {}, state = null) => {
    const wrapper = shallow(<Component {...props} />);

    if (state) {
        wrapper.setState(state);
    }

    return wrapper;
};

/**
 * Return ShallowWrapper containg node(s) with the data-test value
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within
 * @param {string} value - data attribute value to find
 * @returns {ShallowWrapper}
 */
export const findComponent = (wrapper, value) => wrapper.find(`[data-test="${value}"]`);

/**
 * Check corret properties in component
 * @param {ReactComponent} component
 * @param {object} props
 * @returns {Error|undefined}
 */
export const checkProps = (component, props) => checkPropTypes(component.propTypes, props, 'props', component.name);
