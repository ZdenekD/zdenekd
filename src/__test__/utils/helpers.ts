import {ShallowWrapper} from 'enzyme';

const findComponent = (wrapper: ShallowWrapper, attribute: string): ShallowWrapper => wrapper.find(`[data-test="${attribute}"]`);

export default findComponent;
