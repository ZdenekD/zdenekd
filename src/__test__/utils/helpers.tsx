import {
    mount,
    ReactWrapper,
    ShallowWrapper
} from 'enzyme';
import {Provider} from '../../state';

type IWrapper = ShallowWrapper | ReactWrapper;

const findComponent = (wrapper: IWrapper, attribute: string): IWrapper => wrapper.find(`[data-test="${attribute}"]`);

export const stateFactory = (children: React.ReactNode): IWrapper => {
    const wrapper = mount(
        <Provider>
            {children}
        </Provider>
    );

    return wrapper;
};

export default findComponent;
