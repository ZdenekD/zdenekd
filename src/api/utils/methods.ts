import {IMethods, IOptions} from '@/api/types';

const headers: HeadersInit = {
    Accept: 'application/json',
    'Content-Type': 'application/json; charset=utf-8',
};
const getHeaders = (options?: IOptions): HeadersInit => (
    options?.token
        ? {
            ...headers,
            authorization: `Bearer ${options.token}`,
            'Content-language': options?.language || 'cs',
        }
        : {
            ...headers,
            'Content-language': options?.language || 'cs',
        }
);
const methods: IMethods = {
    get: options => ({
        method: 'GET',
        headers: getHeaders(options),
        signal: options?.signal,
    }),
    post: (data, options) => ({
        method: 'POST',
        headers: getHeaders(options),
        signal: options?.signal,
        body: data ? JSON.stringify(data) : null,
    }),
    put: (data, options) => ({
        method: 'PUT',
        headers: getHeaders(options),
        signal: options?.signal,
        body: JSON.stringify(data),
    }),
    patch: (data, options) => ({
        method: 'PATCH',
        headers: getHeaders(options),
        signal: options?.signal,
        body: JSON.stringify(data),
    }),
    delete: options => ({
        method: 'DELETE',
        headers: getHeaders(options),
        signal: options?.signal,
    }),
};

export default methods;
