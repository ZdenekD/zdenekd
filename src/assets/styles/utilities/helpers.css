:global ._visuallyhidden {
    width: 1px;
    height: 1px;
    margin: -1px;
    padding: 0;
    position: absolute;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    border: none;
}

:global ._align-right {
    text-align: right !important;
}
