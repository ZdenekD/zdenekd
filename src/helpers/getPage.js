import data from '../data/pages';

/**
 * Get page data object to work with in component
 * @param {string} link - URL link value to find in data object
 * @returns {string}
 */
const getPage = link => Object.keys(data).filter(item => data[item].link === link)[0];

/**
 * Get page index from data object
 * @param {string} link - URL link value to find index
 * @returns {number}
 */
const getPageIndex = link => Object.keys(data).findIndex(item => data[item].link === link);

export {
    getPage,
    getPageIndex
};
