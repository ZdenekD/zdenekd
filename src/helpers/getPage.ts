import pages from '../data/pages';

const getPage = (link: string): string => Object.keys(pages).filter(item => `/${pages[item].slug}` === link)[0];

const getPageIndex = (link: string): number => Object.keys(pages).findIndex(item => `/${pages[item].slug}` === link);

export {
    getPage,
    getPageIndex
};
