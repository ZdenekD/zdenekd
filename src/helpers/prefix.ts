const prefix = '_34rFI';

export default prefix;
