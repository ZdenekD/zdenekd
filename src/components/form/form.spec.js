import {factoryState, findComponent} from '../../__test__/utils/helpers';
import Form from './index';

describe('Form', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = factoryState(Form);
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-form');

        expect(component.exists()).toBe(true);
    });
});
