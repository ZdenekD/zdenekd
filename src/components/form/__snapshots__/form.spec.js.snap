// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form match snapshot 1`] = `
<form data-test="component-form">
  <div class="animated-block">
    <div class="undefined ">
      <span class="undefined ">
        0 / 32
      </span>
      <input id="Jméno"
             class="undefined  "
             data-test="component-input"
             required
             name="name"
             type="text"
             maxlength="32"
             data-pattern="name"
             value
      >
      <label for="Jméno"
             class="undefined "
             data-test="component-label"
      >
        <span>
          Jméno
        </span>
      </label>
    </div>
  </div>
  <div class="animated-block">
    <div class="undefined ">
      <span class="undefined ">
        0 / 64
      </span>
      <input id="E-mail"
             class="undefined  "
             data-test="component-input"
             required
             name="email"
             type="email"
             maxlength="64"
             data-pattern="email"
             value
      >
      <label for="E-mail"
             class="undefined "
             data-test="component-label"
      >
        <span>
          E-mail
        </span>
      </label>
    </div>
  </div>
  <div class="undefined ">
    <input id="Med"
           data-test="component-input"
           name="med"
           type="text"
           autocomplete="off"
    >
    <label for="Med"
           class="undefined "
           data-test="component-label"
    >
      <span>
        Med
      </span>
    </label>
  </div>
  <div class="animated-block">
    <div>
      <span class="undefined ">
        0 / 140
      </span>
      <textarea id="Vzkaz"
                class="undefined  "
                data-test="component-textarea"
                required
                name="message"
                maxlength="140"
                data-pattern="text"
                style="height: 0px;"
      >
      </textarea>
      <label for="Vzkaz"
             class="undefined "
             data-test="component-label"
      >
        <span>
          Vzkaz
        </span>
      </label>
    </div>
  </div>
  <div class="animated-block">
    <div class="undefined undefined">
      <span>
      </span>
      <button data-test="component-button"
              type="submit"
              aria-label="Odeslat formulář"
      >
        Odeslat
      </button>
    </div>
  </div>
</form>
`;
